python/ray/data/_internal/iterator/iterator_impl.py

```diff
@@ -58,4 +58,7 @@ def __getattr__(self, name):
         raise AttributeError()
 
     def _get_dataset_tag(self):
-        return (self._base_dataset._plan._dataset_name or "") + self._base_dataset._uuid
+        return (
+            f"{self._base_dataset._plan._dataset_name or 'dataset'}"
+            f"_{self._base_dataset._uuid}"
+        )
```

Comment:
let's define a util function for generating the tag? it's repeated multiple times.